### PR TITLE
change CMDSTAN version for Apple M2

### PR DESCRIPTION
**Pull request by lucentcosmos**
This is needed to build prophet on Apple M2 Pro. 
```
% pip install prophet -e "git+https://github.com/lucentcosmos/prophet.git@441d77e#egg=prophet&subdirectory=python"

Obtaining prophet from git+https://github.com/lucentcosmos/prophet.git@441d77e#egg=prophet&subdirectory=python

  Cloning https://github.com/lucentcosmos/prophet.git (to revision 441d77e) to .....
  WARNING: Did not find branch or tag '441d77e', assuming revision or ref.
  Running command git checkout -q 441d77e
  Resolved https://github.com/lucentcosmos/prophet.git to commit 441d77e
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... done
  Preparing editable metadata (pyproject.toml) ... done
Collecting prophet
..........
Building wheels for collected packages: prophet
  Building editable for prophet (pyproject.toml) ... done
  Created wheel for prophet: filename=prophet-1.1.4a0-0.editable-py3-none-macosx_13_0_arm64.whl size=4722 sha256=33ad80ae8bd2993bff5dc775675e3be5ca6797466ee09d7eed7b32b99bbcc122
  Stored in directory: /private/var/folders/d5/wjs2_3q13f31k5xnnc_t_1bc0000gq/T/pip-ephem-wheel-cache-iv4bmayp/wheels/4d/14/7b/6942cc1367b12354825984ff85aacc56d594630a89181f54cb
Successfully built prophet
Installing collected packages: prophet
  Attempting uninstall: prophet
    Found existing installation: prophet 1.1.4a0
    Uninstalling prophet-1.1.4a0:
      Successfully uninstalled prophet-1.1.4a0
Successfully installed prophet-1.1.4a0

```

### CODE REVIEW

File: python/prophet/models.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def sampling(self, stan_init, stan_data, samples, **kwargs) -> dict:
 
 
 class CmdStanPyBackend(IStanBackend):
-    CMDSTAN_VERSION = "2.31.0"
+    CMDSTAN_VERSION = "2.33.0"
```
Comment (nicolaerosia) [inline python/prophet/models.py:57]: latest is 2.33.1, https://github.com/stan-dev/cmdstan/releases

Comment (lucentcosmos) [reply python/prophet/models.py]: updated and tested on my M2

**Comment by WardBrian** [comment]
Note that 2.32.1 should be good enough: https://github.com/stan-dev/cmdstan/issues/1158

@tcuongd I recommend merging this if CI passes, it should fix the remaining issues in #2503, #2438, and maybe even #2402

**Comment by tcuongd** [comment]
The RMSE in one of the tests is now a bit off:

```
FAILED prophet/tests/test_prophet.py::TestProphetFitPredictDefault::test_fit_predict[CMDSTANPY-minmax] - AssertionError: backend: <prophet.models.CmdStanPyBackend object at 0x11dc0e160>
assert 11.917838773340984 == 11.13 ± 2.2e-01
  comparison failed
  Obtained: 11.917838773340984
  Expected: 11.13 ± 2.2e-01
```

I don't think this is a big deal especially since the other tests have passed, but @lucentcosmos could you do some quick fit / predict runs on your machine and check that the forecasts aren't doing something weird? If it's just down to noise then we can update the expected rmse value in the tests.

**Comment by tcuongd** [comment]
Actually now that I think about it, it's weird that the test only fails for macOS given that it's seeded. We'd need to put a higher tolerance on the check then, if the fitting is actually working as expected.

**Comment by WardBrian** [comment]
I believe this range of updates includes when we updated to Eigen 3.4, which did lead to some numerical changes on some architectures

**Comment by tcuongd** [comment]
Keen to get this merged, but the test result is still weird to me. Just to correct myself from before, the discrepancy is actually happening on Intel Macs; I tested on my M1 and the result is ~11.04 with the same seeds which is fine. If anyone can test on an Intel that the forecast isn't doing anything silly that would be great. Otherwise I'll increase the tolerance to 0.1 (10% relative) and leave a note to revisit.

**Pull request merged**